src/index.ts

```diff
@@ -0,0 +1,201 @@
+import { generateSalt, digest } from './crypto';
+import { Jwt } from './jwt';
+import { KBJwt } from './kbjwt';
+import { SDJwt, pack } from './sdjwt';
+import {
+  DisclosureFrame,
+  KB_JWT_TYP,
+  SDJWTCompact,
+  SDJWTConfig,
+  SD_JWT_TYP,
+  kbPayload,
+} from './type';
+import { KeyLike } from 'jose';
+
+export * from './type';
+export * from './sdjwt';
+export * from './kbjwt';
+export * from './crypto';
+export * from './jwt';
+export * from './base64url';
+export * from './decoy';
+export * from './disclosure';
+
+export const defaultConfig: Required<SDJWTConfig> = {
+  omitTyp: false,
+  hasher: digest,
+  saltGenerator: generateSalt,
+};
+
+export class SDJwtInstance {
+  public static DEFAULT_ALG = 'EdDSA';
+  public static DEFAULT_HASH_ALG = 'sha-256';
+
+  private userConfig: SDJWTConfig = {};
+
+  constructor(userConfig?: SDJWTConfig) {
+    if (userConfig) {
+      this.userConfig = userConfig;
+    }
+  }
+
+  public create(userConfig?: SDJWTConfig): SDJwtInstance {
+    return new SDJwtInstance(userConfig);
+  }
+
+  private async createKBJwt(
+    payload: kbPayload,
+    privateKey: Uint8Array | KeyLike,
+    alg: string,
+  ): Promise<KBJwt> {
+    const kbJwt = new KBJwt({
+      header: {
+        typ: KB_JWT_TYP,
+        alg,
+      },
+      payload,
+    });
+    await kbJwt.sign(privateKey);
+    return kbJwt;
+  }
+
+  public async issue<Payload extends object>(
```

Comment:
Main functions like issue and verify should be separated as individual files to help out navigation

Reply:
I think it will be needed as there are more files. I will proceed then.